resources/group_snippet.js

```diff
@@ -45,44 +39,23 @@ class Client {
 }
 
 function trimAndRemovePrefix(input) {
-  // Use the trim method to remove leading and trailing whitespace
-  let trimmedInput = input.trim()
-
-  // Check if the trimmed input starts with "::" and remove it
-  if (trimmedInput.startsWith('::')) {
-    trimmedInput = trimmedInput.slice(2)
-  }
-
-  return trimmedInput
+  return input.trim().replace(/^::/, '')
 }
 
-function instrumentArguments(groupName, ...args) {
-  switch (args.length) {
-    case 0:
-      args.push(null)
-    // fallthrough to add the header
-    case 1:
-      // We only received a body argument
-      args.push({ headers: groupName })
-      break
-    default: // this handles 2 and more just in case someone provided more arguments
-      // We received both a body and a params argument. In the
-      // event params would be nullish, we'll instantiate
-      // a new object.
-      if (args[1] == null) args[1] = {}
-
-      let params = args[1]
-      if (params.headers == null) {
-        params.headers = {}
-      }
-      Object.assign(params.headers, groupName)
-      break
+function instrumentArguments(args) {
+  const [body = null, params = {}] = args
+  const groupHeaders = {
+    'X-k6-group': trimAndRemovePrefix(execution.vu.tags.group),
   }
 
-  return args
+  const updatedParams = Object.assign({}, params, {
+    headers: Object.assign({}, params.headers || {}, groupHeaders),
+  })
```

Comment:
TIL that our version of k6 doesn't support the spread operator. It was added to a newer version (https://github.com/grafana/k6/pull/3456)